go/libraries/doltcore/schema/super_schema.go

```diff
@@ -69,6 +69,14 @@ func (ss *SuperSchema) AddColumn(col Column) (err error) {
 			return fmt.Errorf("tag collision for columns %s and %s, different definitions (tag: %d)",
 				ecName, col.Name, col.Tag)
 		}
+
+		// If the column's primary key metadata has changed, update SuperSchema's
+		// representation of the column to match.
+		if existingCol.IsPartOfPK != col.IsPartOfPK {
```

Comment:
Seems like this is unrelated to the primary purpose of this PR, right? Could it be its own PR, or is this a bug that affects tests you're writing?

Reply:
Yup, it came out as part of testing the DOLT_BLAME view. It isn't directly related to the DOLT_BLAME functionality, but it was causing my tests to fail because of how they were written (altering PKs to test different schemas). This definitely felt like it needed more testing, so I put it in its own commit; it'd be easy to pull it out to another branch/change/PR and look at it separately.

Reply:
Removed this commit from this PR and pushed to a new branch (fulghum/superschema-primarkey-bugfix). I'll circle back to that after getting the DOLT_BLAME view out.